src/app/pages/listings/contents/masonary.directive.ts

```diff
@@ -0,0 +1,45 @@
+import { Directive, ElementRef, OnDestroy, OnInit } from '@angular/core';
+
+@Directive({
+  selector: '[appMasonary]',
+})
+export class MasonaryDirective implements OnInit, OnDestroy {
+  private observer!: MutationObserver;
+
+  constructor(private gridElement: ElementRef) {}
+
+  get cardElments() {
+    return Array.from(this.gridElement.nativeElement.children);
+  }
+
+  ngOnInit() {
+    this.observer = new MutationObserver(() => {
+      this.cardElments.forEach((cardElement) => {
+        this.resizeMasonryItem(cardElement as HTMLElement);
+      });
+    });
+    this.observer.observe(this.gridElement.nativeElement, { childList: true });
+  }
+
+  //Reference:https://w3bits.com/css-grid-masonry/
+  resizeMasonryItem = (cardeElment: HTMLElement) => {
+    /* Get the grid object, its row-gap, and the size of its implicit rows */
+    const grid = this.gridElement.nativeElement;
+    const rowGap = parseInt(
+      window.getComputedStyle(grid).getPropertyValue('grid-row-gap'),
+      10
+    );
+    const rowHeight = 0;
+    const rowSpan = Math.ceil(
+      (cardeElment.querySelector('.listing-card')!.getBoundingClientRect()
+        .height +
+        rowGap) /
+        (rowHeight + rowGap)
+    );
+    cardeElment.style.gridRowEnd = `span ${rowSpan}`;
+  };
```

Comment:
Can you explain this in the feature request you created so that I'm able to write update on it